examples/scala-parallel-similarproduct/filterbyyear/src/main/scala/DataSource.scala

```diff
@@ -13,16 +16,21 @@ import org.apache.spark.rdd.RDD
 
 import grizzled.slf4j.Logger
 
-case class DataSourceParams(appId: Int) extends Params
+case class DataSourceParams(appName: String, eventWindow: Option[EventWindow], appId: Int) extends Params
 
 class DataSource(val dsp: DataSourceParams)
   extends PDataSource[TrainingData,
-      EmptyEvaluationInfo, Query, EmptyActualResult] {
+      EmptyEvaluationInfo, Query, EmptyActualResult] with SelfCleaningDataSource {
+
+  @transient override lazy val logger = Logger[this.type]
 
-  @transient lazy val logger = Logger[this.type]
+  override def appName = dsp.appName
+  override def eventWindow = dsp.eventWindow
 
   override
   def readTraining(sc: SparkContext): TrainingData = {
+    val events = cleanPersistedPEvents(sc) 
+
     val eventsDb = Storage.getPEvents()
 
     // create a RDD of (entityID, User)
```

Comment:
ready for review

adds vals needed by the SelfCleaningDatasource trait as an example and for tests.